contracts/OptionLib.sol

```diff
@@ -94,7 +94,22 @@ contract OptionLib is UniswapV2Router02 {
         );
 
         // mint options and obligations - locking collateral
-        IObligation(obligation).mint(msg.sender, optionsAmount);
+        IObligation(obligation).mintToPool(msg.sender, optionsAmount);
+    }
+
+    /// @notice Atomically deposit collateral into a treasury and directly credit
+    /// the writer with the resulting options.
+    /// @param obligation The obligation tokens
+    /// @param collateral The collateral tokens
+    /// @param optionsAmount The amount of options that will be minted (hence the amount of collateral that must be locked)
+    function lockAndWriteToSelf(
```

Comment:
```suggestion
    function lockAndWriteToWriter(
```

Reply:
Let's equate the naming conventions, can you also rename the one above to `lockAndWriteToPool`?

Reply:
My reasoning was that it credits msg.sender as the writer, but you definitely have a point with keeping to convention, updating